thunder/torch/__init__.py

```diff
@@ -561,12 +561,31 @@ def arange(
     return clang.arange(start=start, step=step, stop=end, device=device, dtype=dtype)
 
 
+# Infers dtype from the fill_value and dtype
+def _infer_full_dtype(fill_value: NumberLike, dtype):
+    fill_value_dtype = dtypes.numbertype_to_dtype(dtypes.to_dtype(fill_value))
+    if not dtype:
+        current_default_dtype = get_default_dtype()
+        if dtypes.is_boolean_dtype(fill_value_dtype):
+            return fill_value_dtype
+        elif dtypes.is_nonboolean_integer_dtype(fill_value_dtype):
+            return dtypes.int64
+        # NOTE Thunder has slightly different behavior than Torch, Torch (2.5.0a0+git8927fc2):
```

Comment:
Would you elaborate on this comment about what is PyTorch's behavior, and what is meant by Thunder's behavior?

Is the behavior different in this function, or is the difference elsewhere?

Reply:
It's updated in the new commit: https://github.com/Lightning-AI/lightning-thunder/pull/949/commits/bc3358994e833eca15a571cdd44434211679fec7